gitpod-io/gitpod
# [server] Introduce EntitlementServiceChargbee and move relevant parts of EligibilityService into it

## Description
This PR does two things, while **not touching current behavior**:
 - it introduces the `EntitlementService` interface, which is based on a subset of methods of `EligibilityService`
 - it add a Chargebee implementation, which is 100% the identical code copied over from `EligibilityService`

## Related Issue(s)
Context: https://github.com/gitpod-io/gitpod/issues/11402

## How to test
 - open a workspace on this PR
 - get the content of the `EligiblityService` [before this commit](https://raw.githubusercontent.com/gitpod-io/gitpod/e3452c71b2fe1e81bfb91b5cc25bc8089d4efbde/components/server/ee/src/user/eligibility-service.ts), and put it into a file called `eligibilityservice.ts`
 - run `diff ee/src/billing/entitlement-service-chargebee.ts eligibilityservice.ts | less`
 - inspect the output carefully; notice how there is no relevant diff between the two

## Release Notes

```release-note
NONE
```

## Documentation


## Werft options:
